### PR TITLE
Remove jupyter & visualization modules

### LINKED ISSUE #112: Revive IQX dashboard widget
**What is the expected feature or enhancement?**

In qiskit-ibmq-provider there was an IQX dashboard that displayed backends as well as legacy jobs. It'd be nice to have something similar, especially for cloud runtime since they don't show up on IQX. 

**Acceptance criteria**

A user can see runtime backend and job information on a dashboard.

### LINKED ISSUE #124: Revive or discontinue HTML representation for IBM Backends
**What is the expected feature or enhancement?**
In qiskit-ibmq-provider users could opt-in to HTML representation of IBM backends by importing `qiskit.providers.ibmq.jupyter`
```python
# cf. https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/docs/tutorials/2_jupyter_tools.ipynb
import qiskit.providers.ibmq.jupyter  # This is the where the magic happens (literally).
from qiskit.providers.ibmq import least_busy

backend = least_busy(...)
backend
```
As a result the backend information was displayed as an HTML widget
image

The related code has been ported to `qiskit-ibm-runtime` but not been updated to work with latest code changes. Therefore displaying a backend fails when `qiskit_ibm_runtime.jupyter` is imported. As this is not a documented API, it's not an issue that users are likely to hit.

```python
import qiskit_ibm_runtime.jupyter
backend = # ...
backend
```

**Acceptance criteria**
- Option 1 - HTML representation of IBM backends is documented, tested and works
- Option 2 - HTML representation of IBM backends is discontinued, dead code paths have been removed

### PR DESCRIPTION
**Pull request by kt474**
### Details and comments
Fixes #112, Fixes #124

### CODE REVIEW

**Review by daka1510 — CHANGES_REQUESTED** [review]
Looks good in general. 

I think we can clean up a bit more related lines
- https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/setup.py#L77-L87
- https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/requirements-dev.txt

Can you check if you can remove development dependencies that were only used by the removed packages? Thinking of `vuetify` but there might be more...

**Comment by kt474** [comment]
Removing `constraints.txt` now that the last constraint, `traitlets` is no longer used

File: tox.ini [hunk 1]
```diff
@@ -5,7 +5,6 @@ skipsdist = True
 
 [testenv]
 usedevelop = true
-install_command = pip install -c{toxinidir}/constraints.txt -U {opts} {packages}
```
Comment (rathishcholarajan) [inline tox.ini:8]: Why is this entire command removed?

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
Mostly looks good...

File: tox.ini (hunk 1 above)
Comment (kt474) [reply tox.ini:8]: Missed the `install_command` syntax in [tox](https://tox.wiki/en/latest/config.html) - fixed now

**Review by daka1510 — APPROVED** [review]
Looks good to me now, thanks @kt474.

**Comment by daka1510** [comment]
And that sets a high bar - increased test coverage by 13% with a single PR 😆.

**Pull request merged**